python.d/spigotmc.chart.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# Description: spigotmc netdata python.d module
+# Author: Austin S. Hemmelgarn (Ferroin)
+
+import socket
+
+from bases.FrameworkServices.SimpleService import SimpleService
+
+from third_party import mcrcon
+
+PRECISION = 100
+
+ORDER = ['tps', 'users']
+
+CHARTS = {
+    'tps': {
+        'options': [None, 'Spigot Ticks Per Second', 'ticks', 'spigotmc', 'spigotmc.tps', 'line'],
+        'lines': [
+            ['tps1', '1 Minute Average', 'absolute', 1, PRECISION],
+            ['tps5', '5 Minute Average', 'absolute', 1, PRECISION],
+            ['tps15', '15 Minute Average', 'absolute', 1, PRECISION]
+        ]
+    },
+    'users': {
+        'options': [None, 'Minecraft Users', 'users', 'spigotmc', 'spigotmc.users', 'area'],
+        'lines': [
+            ['users', 'Users', 'absolute', 1, 1]
+        ]
+    }
+}
+
+class Service(SimpleService):
+    def __init__(self, configuration=None, name=None):
+        SimpleService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.host = self.configuration.get('host', 'localhost')
+        self.port = self.configuration.get('port', 25575)
+        self.password = self.configuration.get('password', '')
+        self.console = mcrcon.MCRcon()
+
+    def check(self):
+        '''Check plugin configuration validity.
+
+           This really just makes sure we can connect and authenticate
+           correctly.'''
+        try:
+            self.console.connect(self.host, self.port, self.password)
+        except (mcrcon.MCRconException, socket.error):
+            return False
+        return True
+
+    def _get_data(self):
+        data = {}
+        try:
+            raw = self.console.command('tps')
+            # The above command returns a string that looks like this:
+            # '§6TPS from last 1m, 5m, 15m: §a19.99, §a19.99, §a19.99\n'
+            # The values we care about are the three numbers after the :
+            tmp = raw.split(':')[1].split(',')
+            data['tps1'] = float(tmp[0].lstrip()[2:]) * PRECISION
+            data['tps5'] = float(tmp[1].lstrip()[2:]) * PRECISION
+            data['tps15'] = float(tmp[2].lstrip().rstrip()[2:]) * PRECISION
+        except (mcrcon.MCRconException, socket.error):
+            data['tps1'] = None
+            data['tps5'] = None
+            data['tps15'] = None
```

Comment:
Not sure these `None` makes any sense

Reply:
I was under the impression from looking at the other Python plugins that returning `None` says that that value couldn't be collected.  It doesn't make much sense here to report anything if we failed (reporting 0 would be potentially misleading).

Reply:
just don't add a key if it is missing

Reply:
Ah, OK, that makes sense.